.buildkite/commands/save-cache.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -euo pipefail
+
+echo "--- :rubygems: Setting up Gems"
+install_gems
+
+echo "--- :closed_lock_with_key: Installing Secrets"
+bundle exec fastlane run configure_apply
+
+echo "--- 🛠 Download Mobile App Dependencies [Assemble Mobile App]"
+./gradlew :WooCommerce:assembleJalapenoDebug
+echo ""
+
+echo "--- 🛠 Download Wear App Dependencies [Assemble Wear App]"
+./gradlew :WooCommerce-Wear:assembleJalapenoDebug
+echo ""
+
+echo "--- 🧹 Download Lint Dependencies [Lint Mobile App]"
+./gradlew :WooCommerce:lintJalapenoDebug
+echo ""
+
+echo "--- 🧪 Download Unit Test Dependencies [Assemble Unit Tests]"
+./gradlew assembleJalapenoDebugUnitTest lib:cardreader:assembleDebugUnitTest
```

Comment:
This will not cache dependencies for all submodules, I think we want rather this 
```suggestion
./gradlew assembleJalapenoDebugUnitTest assembleDebugUnitTest
```

Reply:
Oh, thanks for noticing that @wzieba , I was trying to have this on a 1-on-1 relationship with this corresponding [run-unit-tests.sh](https://github.com/woocommerce/woocommerce-android/blob/trunk/.buildkite/commands/run-unit-tests.sh) script, but it was recently changed (758606435bdb7287b8c680a46a005ab15f723823) as part of this #13052 PR into exactly [that](https://github.com/woocommerce/woocommerce-android/blob/trunk/.buildkite/commands/run-unit-tests.sh#L5), I'll proceed with the change. 💯 

PS: Btw, for some reason I was thinking, and I could actually swear that I tested it in the past, that running `assembleDebugUnitTest` will also trigger `assembleJalapenoDebugUnitTest`, plus `assembleWasabiDebugUnitTest` and `assembleVanillaDebugUnitTest`, because of all the different variants, but actually, me testing it locally again, it doesn't looks like it. 👍

Reply:
Done: cbca6ac19fcc9320bccaac54d2a4dcd227a4fbb1